library/src/app/components/transfer/transfer-confirm/transfer-confirm.component.spec.ts

```diff
@@ -0,0 +1,235 @@
+    );
+    MockQuotesService = TestBed.inject(QuotesService);
+    MockQuotesService.createQuote.and.returnValue(
+      of(TestConstants.QUOTE_BANK_MODEL_TRANSFER)
+    );
+    MockTransfersService = TestBed.inject(TransfersService);
+    MockTransfersService.createTransfer.and.returnValue(
+      of(TestConstants.TRANSFER_BANK_MODEL)
+    );
+    MockTransfersService.getTransfer.and.returnValue(
+      of(TestConstants.TRANSFER_BANK_MODEL)
+    );
+    MockDialog = TestBed.inject(MatDialogRef);
+    MockDialog.open.and.returnValue({
+      afterClosed: () => of(TestConstants.TRANSFER_BANK_MODEL)
+    });
+
+    fixture = TestBed.createComponent(TransferConfirmComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should getQuote()', fakeAsync(() => {
+    component.getQuote();
+
+    tick(Constants.PLATFORM_REFRESH_INTERVAL);
+    expect(MockQuotesService.createQuote).toHaveBeenCalled();
+
+    discardPeriodicTasks();
+  }));
+
+  it('should handle errors on getQuote()', fakeAsync(() => {
+    MockQuotesService.createQuote.and.returnValue(error$);
+
+    component.getQuote();
+
+    tick(Constants.PLATFORM_REFRESH_INTERVAL);
+    expect(MockDialog.close).toHaveBeenCalled();
+    expect(MockSnackbar.open).toHaveBeenCalled();
+    expect(MockErrorService.handleError).toHaveBeenCalled();
+    expect(MockEventService.handleEvent).toHaveBeenCalled();
+  }));
+
+  it('should create transfer onConfirmTransfer()', () => {
+    const isLoadingSpy = spyOn(component.isLoading$, 'next');
+
+    component.onConfirmTransfer();
+
+    expect(isLoadingSpy).toHaveBeenCalledWith(true);
+    expect(MockTransfersService.createTransfer).toHaveBeenCalled();
+  });
+
+  it('should get transfer onConfirmTransfer()', fakeAsync(() => {
+    let transfer = { ...TestConstants.TRANSFER_BANK_MODEL };
+    transfer.state = 'storing';
+
+    // Mock 'state' as 'storing' for create call
+    MockTransfersService.createTransfer.and.returnValue(of(transfer));
+
+    component.onConfirmTransfer();
+
+    tick();
+    expect(MockTransfersService.getTransfer).toHaveBeenCalled();
+    expect(MockDialog.close).toHaveBeenCalled();
+  }));
+
+  it('should poll on getTransfer()', fakeAsync(() => {
+    let transfer = { ...TestConstants.TRANSFER_BANK_MODEL };
+    transfer.state = 'storing';
+
+    // Mock 'state' as 'storing' for both create and get calls
+    MockTransfersService.createTransfer.and.returnValue(of(transfer));
+    MockTransfersService.getTransfer.and.returnValue(of(transfer));
+
+    // Reset spy
+    MockTransfersService.getTransfer.calls.reset();
+
+    component.onConfirmTransfer();
+    tick(Constants.POLL_DURATION);
+
+    const callCount = Constants.POLL_DURATION / Constants.POLL_INTERVAL;
+    expect(MockTransfersService.getTransfer).toHaveBeenCalledTimes(callCount);
+  }));
+
+  it('should handle a timeout on getTransfer()', () => {
```

Comment:
Transfers can take days to complete. Is the component dependent on the transfer being completed in a short time period?

Reply:
As long as the transfer passes 'storing' it will continue through the flow and show the transfer details (including state).